helpers/updateSubmission.ts

```diff
@@ -14,12 +15,60 @@ export const updateSubmission = async (
     if (!args) throw new Error('Invalid args')
     const { id, comment, status, reviewerId } = args
     const submission = await Submission.findByPk(id)
+    const challengeCount = await Challenge.count({
+      where: { lessonId: submission.lessonId }
+    })
 
     submission.set('reviewerId', reviewerId)
     submission.set('status', status)
     submission.set('comment', comment)
     await submission.save()
 
+    const submissions = await Submission.findAll({
+      where: {
+        lessonId: submission.lessonId,
+        userId: submission.userId
+      }
+    })
+
+    const passedSubmissions = submissions.reduce(
+      (sum: number, s: any) => sum + (s.status === 'passed' ? 1 : 0),
+      0
+    )
+
+    if (challengeCount === passedSubmissions) {
+      const userLesson = await UserLesson.findOne({
+        where: { lessonId: submission.lessonId }
+      })
+
+      if (!userLesson.isPassed) {
+        const { email } = await User.findByPk(submission.userId)
+        const { order, title, chatUrl } = await Lesson.findByPk(
+          userLesson.lessonId
+        )
+        const chatUsername = await getUserByEmail(email)
+        const nextLesson = await Lesson.findOne({
+          where: { order: order + 1 }
```

Comment:
should be `previousLesson` :)

If I'm in JS0, my submissions goes to JS1. so... when my last JS0 submission is approved, current channel (JS1) will be updated that I'm joining. And then previous lesson (JS0) should be notified that I passed. 

Add a comment for this too!

Reply:
But the submission that's being updated belongs to the `currentLesson` and the `nextLesson` would be the channel that the user is joining right?

If I'm currently in JS0 and my last submission for JS0 get's approved, my currentLesson is still JS0 at that point in time, and my next lesson would be JS1.

Reply:
`currentLesson.chatUrl` is actually the chatroom for the next lesson.